src/coreclr/jit/lower.cpp

```diff
@@ -7752,6 +7777,66 @@ GenTree* Lowering::LowerSignedDivOrMod(GenTree* node)
     return node->gtNext;
 }
 
+//------------------------------------------------------------------------
+// TryFoldBinop: Try removing a binop node by constant folding.
+//
+// Parameters:
+//   node - the node
+//
+// Returns:
+//   True if the node was removed
+//
+bool Lowering::TryFoldBinop(GenTreeOp* node)
+{
+    if (node->gtSetFlags())
+    {
+        return false;
+    }
+
+    GenTree* op1 = node->gtGetOp1();
+    GenTree* op2 = node->gtGetOp2();
+
+    if (op1->IsIntegralConst() && op2->IsIntegralConst() &&
+        op1->AsIntConCommon()->ImmedValCanBeFolded(comp, node->gtOper) &&
+        op2->AsIntConCommon()->ImmedValCanBeFolded(comp, node->gtOper))
```

Comment:
Do we need these checks? I'd expect gtFoldExprConst to check them

Reply:
Good point, looks like it does -- removed

Reply:
(I kept the const checks because I want some control over when we call this "normally used on HIR" API)